### PR TITLE
Add tests for multiple scheme implementations

### PR DESCRIPTION
**Pull request by graywolf**
This pull request reworks the test infrastructure to allow
simple testing on multiple scheme implementations.  It also fixes
few bugs.

### CODE REVIEW

File: irregex.scm [hunk 1]
```diff
@@ -363,7 +363,14 @@
 ;; (define *all-chars* `(/ ,(integer->char (- (char->integer #\space) 32)) ,(integer->char (+ (char->integer #\space) 223))))
 
 ;; set to #f to ignore even an explicit request for utf8 handling
-(define *allow-utf8-mode?* #t)
+;; The utf8-mode is undesired on any implementation with native unicode support.
+;; It is a workaround for those that treat strings as a raw byte sequences, and
+;; does not work well otherwise.  So disable it on implementations known to
+;; handle unicode natively.
+(define *allow-utf8-mode?* (cond-expand (chibi #f)
```
Comment (ashinn) [inline irregex.scm:370]: I think we can actually do `(cond-expand (chicken #t) (else #f))`

File: manifest.scm [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+             (gnu packages guile)
+             (gnu packages scheme)
+             (guix build-system chicken)
+             (guix download)
+             ((guix licenses) #:prefix license:)
+             (guix packages)
+             (guix profiles)            ; For packages->manifest in the REPL
+             )
+
+(define-public chicken-matchable
```
Comment (ashinn) [inline manifest.scm:11]: Why is matchable here?

File: tests/chicken/test-support.scm [hunk 3]
```diff
@@ -0,0 +1,4 @@
+(define-syntax test-equal
```
Comment (ashinn) [inline tests/chicken/test-support.scm:1]: Both the Chicken test egg and Chibi test libraries use `test`, I prefer to keep that.

File: test-irregex-gauche.scm [hunk 4]
```diff
@@ -1,1001 +0,0 @@
-
-;; derived from Gauche's regex tests
```
Comment (ashinn) [inline test-irregex-gauche.scm:2]: So this is _not_ a version of the test suite ported to Gauche, but the Gauche native regex test suite ported to irregex, i.e. these tests are not duplicated in any other files so should be preserved.

File: tests/chibi-irregex.scm [hunk 5]
```diff
@@ -0,0 +1,10 @@
+(import (chibi match)
+        (chibi test)
+        (scheme small)
+        (scheme r5rs)
+        (srfi 130))
+
+(load "irregex.scm")
```
Comment (ashinn) [inline tests/chibi-irregex.scm:7]: nit: We should use modules for implementations which support R7RS but that can be done as a follow-up.

**Review by ashinn — CHANGES_REQUESTED** [review]
Thanks!  Sorry for the delay, I literally just moved.

File: tests/chicken/test-support.scm (hunk 3 above)
Comment (sjamaan) [reply tests/chicken/test-support.scm:1]: CHICKEN core uses `test-equal` and every time we update Irregex the tests have to be "ported", so I'd be fine with this change.

File: irregex.scm (hunk 1 above)
Comment (sjamaan) [reply irregex.scm:370]: Make that `(cond-expand ((and chicken (not full-unicode)) #t) (else #f))`

The `full-unicode` feature is (or will be) available in the upcoming CHICKEN 6.

Comment (ashinn) [reply irregex.scm:370]: :tada:

File: tests/chicken/test-support.scm (hunk 3 above)
Comment (ashinn) [reply tests/chicken/test-support.scm:1]: Why not just use this definition, swapping `test` and `test-equal`?

**Comment by ashinn** [comment]
> I believe I resolved everything except the (test) vs (test-equal), where I am waiting for final judgement regarding which way to go.

General manners for a first contribution would be to provide as small a diff as possible :)

Regardless, I like using just `test`.  It's the most common test operation and so deserves a short name, even if we need to rebind it locally.

**Pull request closed without merging**